common/warp_utils/src/json.rs

```diff
@@ -0,0 +1,22 @@
+use bytes::Bytes;
+use serde::de::DeserializeOwned;
+use std::error::Error as StdError;
+use warp::{Filter, Rejection};
+
+use crate::reject;
+
+struct Json;
+
+type BoxError = Box<dyn StdError + Send + Sync>;
+
+impl Json {
+    fn decode<T: DeserializeOwned>(bytes: Bytes) -> Result<T, BoxError> {
+        serde_json::from_slice(&bytes).map_err(Into::into)
+    }
+}
+
+pub fn json<T: DeserializeOwned + Send>() -> impl Filter<Extract = (T,), Error = Rejection> + Copy {
+    warp::body::bytes().and_then(|bytes: Bytes| async move {
+        Json::decode(bytes).map_err(|err| reject::custom_bad_request(format!("{:?}", err)))
+    })
+}
```

Comment:
Diff from `warp::filters::body::json`

```diff
pub fn json<T: DeserializeOwned + Send>() -> impl Filter<Extract = (T,), Error = Rejection> + Copy {
-    is_content_type::<Json>()
        .and(bytes())
        .and_then(|buf| async move {
            Json::decode(buf).map_err(|err| {
                tracing::debug!("request json body error: {}", err);
                reject::known(BodyDeserializeError { cause: err })
            })
        })
}
```

```
impl Decode for Json {
    fn decode<B: Buf, T: DeserializeOwned>(mut buf: B) -> Result<T, BoxError> {
        serde_json::from_slice(&buf.copy_to_bytes(buf.remaining())).map_err(Into::into)
    }
}
```

Reply:
👍  Equivalent code with the suggestion to remove the copy from https://github.com/sigp/lighthouse/pull/4575#discussion_r1430824300

Reply:
@eserilev consider appending some extra message context like

https://github.com/sigp/lighthouse/blob/1be5253610dc8fee3bf4b7a8dc1d01254bc5b57d/common/warp_utils/src/reject.rs#L164-L165